ToolBox/GDAXDownloader/GDAXDownloader.cs

```diff
@@ -0,0 +1,115 @@
+            }
+
+            var counter = startUtc;
+            const string url = "http://api.gdax.com/products/{0}/candles?start={1}&end={2}&granularity={3}";
+
+            while (counter <= endUtc)
+            {
+                System.Threading.Thread.Sleep(500);
+                Log.Trace("Getting " + counter.ToShortDateString() + " data..");
+
+                DateTime endDate = counter.AddDays(1);
+                var granularity = 3600;
+
+                var requestURL = string.Format(url, symbol.Value, counter.ToShortDateString(), endDate.ToShortDateString(), granularity);
+                var request = (HttpWebRequest)WebRequest.Create(requestURL);
+                request.UserAgent = ".NET Framework Test Client";
+
+
+                try
+                {
+                    var response = (HttpWebResponse)request.GetResponse();
+                    var encoding = ASCIIEncoding.ASCII;
+
+                    string data;
+                    using (var reader = new System.IO.StreamReader(response.GetResponseStream(), encoding))
+                    {
+                        data = reader.ReadToEnd();
+                    }
+                    if (data.Length > 0)
+                    {
+                        var a = JArray.Parse(data);
+
+                        foreach (var token in a.Children())
+                        {
+                            var barValues = JArray.Parse(token.ToString());
+
+                            for (int i = 0; i < barValues.Count; i++)
+                            {
+                                var dt = Time.UnixTimeStampToDateTime(double.Parse(barValues[0].ToString()));
+                                var tradeBar = new TradeBar()
+                                {
+                                    Time = dt,
+                                    Symbol = symbol,
+                                    Low = decimal.Parse(barValues[1].ToString()),
+                                    High = decimal.Parse(barValues[2].ToString()),
+                                    Open = decimal.Parse(barValues[3].ToString()),
+                                    Close = decimal.Parse(barValues[4].ToString()),
+                                    Volume = decimal.Parse(barValues[5].ToString()),
+                                    Value = decimal.Parse(barValues[4].ToString()),
+                                    DataType = MarketDataType.TradeBar,
+                                    Period = Time.OneHour,
+                                    EndTime = dt.AddHours(1)
+                                };
+                                returnData.Add(tradeBar);
+                            }
+                        }
+                    }
+                    counter = counter.AddDays(1);
+                }
+                catch(System.Net.WebException ex)
+                {
+                    Console.WriteLine("BAD REQUEST: " + requestURL);

```

Comment:
Log.Error(ex);

Reply:
Generally anything in a re-usable class like this must use Log.X as the users can override the log handlers. Forcing console output in a reusable class means entries won't make it to the system log.

Console.Write etc is only passable in the Program.cs files where its theoretically an "application" of the reusable class object. Even there its better to use Log.Trace to keep it all standardized.

Reply:
Thanks